File: RMS_Flutter/lib/model/food.dart

class Food {
  int? id;
  String? name;
  double? price;
  String? category;
  bool? available;
  String? image;

  Food(
      {this.id,
        this.name,
        this.price,
        this.category,
        this.available,
        this.image});

  Food.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    price = json['price'];
    category = json['category'];
    available = json['available'];
    image = json['image'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['price'] = this.price;
    data['category'] = this.category;
    data['available'] = this.available;
    data['image'] = this.image;
    return data;
  }
}

File: RMS_Spring/src/main/java/com/Project/RMSSpring/entity/Food.java
package com.Project.RMSSpring.entity;

import jakarta.persistence.*;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.*;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "foods")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Food {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(length = 50, nullable = false)
    private String name;

    @Column(length = 50, nullable = false)
    private double price;

    @Column(length = 50, nullable = false)
    private String category;

    @Column(length = 50, nullable = false)
    private boolean available;

    private String image;

}
